pkg/binding/binary_encoder.go

```diff
@@ -0,0 +1,39 @@
+package binding
+
+import (
+	"context"
+	"io"
+
+	"github.com/cloudevents/sdk-go/pkg/binding/spec"
+)
+
+// BinaryEncoder is used to visit a binary Message and generate a new representation.
+//
+// Protocols that supports binary encoding should implement this interface to implement direct
+// binary to binary encoding and event to binary encoding.
+//
+// Start() and End() methods are invoked every time this BinaryEncoder implementation is used to visit a Message
+type BinaryEncoder interface {
+	// Method invoked at the beginning of the visit. Useful to perform initial memory allocations
+	Start(ctx context.Context) error
+
+	// Set a standard attribute.
+	//
+	// The value can either be the correct golang type for the attribute, or a canonical
+	// string encoding. See package types to perform the needed conversions
+	SetAttribute(attribute spec.Attribute, value interface{}) error
```

Comment:
for godoc to work the first word in the comment has to be the function name: `SetAttribute`